tools/assetpipelineeditor/mainwindow.cpp

```diff
@@ -337,6 +344,43 @@ void MainWindow::openFile()
     setFilePath(filePath);
 }
 
+void MainWindow::compressFile()
+{
+    QFile gltfFile(QUrl(m_filePathURL).toLocalFile());
+    if (gltfFile.open(QIODevice::ReadOnly)) {
+        QJsonDocument gltfDocument = QJsonDocument::fromJson(gltfFile.readAll());
+        auto rootObject = gltfDocument.object();
+        auto buffersArray = rootObject["buffers"].toArray();
+        const auto nbBuffers = buffersArray.size();
+
+        QByteArray compressedBufferData;
+        // Meshes are added in to the scene entity as they appear as primitives in the mesh attribute
+        for (const auto &meshName : m_entity->meshes()->names()) {
+            const auto encodedBuffer = compressMesh(*m_entity->mesh(meshName)->geometry());
+            compressedBufferData.push_back(QByteArray { encodedBuffer.get()->data(), static_cast<int>(encodedBuffer.get()->size()) });
+        }
+
+        const QFileInfo gltfFileInfo(m_filePathURL);
+        const QDir gltfFileDir = gltfFileInfo.dir();
+        QFile compressedBufferFile(
+                QUrl(gltfFileDir.filePath({ "compressedBuffer.bin" })).toLocalFile());
+        if (compressedBufferFile.open(QIODevice::WriteOnly))
+            compressedBufferFile.write(compressedBufferData);
+
+        QJsonObject compressedBufferObject;
+        compressedBufferObject["byteLength"] = compressedBufferData.size();
+        compressedBufferObject["uri"] = QUrl(gltfFileDir.filePath({ "compressedBuffer.bin" })).toLocalFile();
```

Comment:
should be relative path

Reply:
Sure, just doing some tests. Now that Ive finally fix the attribute generation, I will address this things